Add documentation on values for the Supply Chain use case (#92)
validatedpatterns/layered-zero-trust: docs/supply-chain.md

* [Multicloud Object Gateway](https://docs.redhat.com/en/documentation/red_hat_openshift_container_storage/4.8/html/managing_hybrid_and_multicloud_resources/index) is a data service for OpenShift that provides an S3-compatible object storage. In our case, this component is necessary to provide a storage system to Quay.
* [Red Hat OpenShift Pipelines](https://docs.redhat.com/en/documentation/red_hat_openshift_pipelines/1.20) is a cloud-native CI/CD solution built on the Tekton framework. We will use this product to automate our secure supply chain process, but you could use your own CI/CD solution if one exists.

### Setting up the values

> We've seen the components we'll use in this use case. To configure them, we must add (uncomment) these parameters to the `values-hub.yaml` file

Reviewer: Should the `values-hub.yaml` be a hyperlink to the file?

> We've seen the components we'll use in this use case. To configure them, we must add (uncomment) these parameters to the `values-hub.yaml` file

Reviewer: What does "We've see the components we'll use in this use case" mean?

Author: I am referring to the components described in the previous sections (_Main_ and _Supporting Components_): RHTAS, RHTPA, ZTWIM, Quay, MCO, and Tekton.

          value: "false"
        - name: rhtpa.modules.createImporters.importers.redhat-sboms.sbom.disabled
          value: "false"
    supply-chain:
      name: supply-chain
      project: hub
      path: charts/supply-chain
      ignoreDifferences:
        - group: ""
          kind: ServiceAccount
          jqPathExpressions:
            - .imagePullSecrets[]|select(.name | contains("-dockercfg-"))
      overrides:
        # Don't forget to uncomment the RHTAS and RHTPA components in this same file
        - name: rhtas.enabled
          value: true
        - name: rhtpa.enabled
          value: true
        - name: registry.tlsVerify
          value: "false"
        - name: registry.user
          value: quay-admin
        - name: registry.passwordVaultKey
          value: quay-admin-password
    ```

* `applications.vault.jwt.roles.policies`: In the **Vault** polices, add the policy **rhtpa**.

    ```shell
          - name: rhtpa
            audience: rhtpa
            subject: spiffe://apps.{{ $.Values.global.clusterDomain }}/ns/trusted-profile-analyzer/sa/rhtpa
            policies:
              - global-secret
    ```

> * `applications.qtodo.overrides`: Configure **qtodo** to use the Quay registry.

Reviewer: ```suggestion
* `applications.qtodo.overrides`: In the **qtodo** overrides section, specify the following to source content from the Quay registry instance.
```

    ```shell
        - name: app.images.main.name
          value: quay-registry-quay-quay-enterprise.apps.{{ $.Values.global.clusterDomain }}/ztvp/qtodo
        - name: app.images.main.version
          value: latest
        - name: app.images.main.registry.auth
          value: true
        - name: app.images.main.registry.user
          value: quay-admin
        - name: app.images.main.registry.passwordVaultKey
          value: quay-admin-password
    ```

> You can apply this changes to your configuration using [this patch file](./values-patches/supply-chain.patch):

Reviewer: ```suggestion
The proceeding changes can also be configured by applying [this patch file](./values-patches/supply-chain.patch):
```

      -o jsonpath='{.items[0].spec.host}')"
  ```

> * Quay username: The same one you specified in `values-hub.yaml` or **quay-admin**.

Reviewer: In my recent testing, `quay-user` is the username that was configured. Not `quay-admin`

Author: You're right. I just fixed it
